common/diagnostics/StatisticReporter.java

```diff
@@ -10,49 +10,117 @@
+            String metricsJSON = metrics.formatJSON(true);
+            metrics.takeSnapshot(); // Forget about this period for consistency even if the https write is not successful
 
-            conn.setRequestMethod("POST");
+            HttpsURLConnection conn = (HttpsURLConnection) (new URL(reportingURI)).openConnection();
 
+            conn.setRequestMethod("POST");
             conn.setRequestProperty("Charset", "utf-8");
             conn.setRequestProperty("Connection", "close");
             conn.setRequestProperty("Content-Type", "application/json");
 
             conn.setDoOutput(true);
-            conn.getOutputStream().write(metrics.formatJSON().getBytes(StandardCharsets.UTF_8));
+            conn.getOutputStream().write(metricsJSON.getBytes(StandardCharsets.UTF_8));
 
             conn.connect();
-
             conn.getInputStream().readAllBytes();
+            return true;
         } catch (Exception e) {
-            if (LOG.isTraceEnabled()) LOG.trace("Failed to push metrics to {}:", reportingURI, e);
-            // do nothing
-        } finally {
-            pushScheduledTask = scheduled.schedule(this::push, 1, HOURS);
+            LOG.warn("Failed to push metrics to {}: {}", reportingURI, e.getMessage());
+            // do nothing and try again after the standard delay
+            return false;
+        }
+    }
+
+    private void scheduleReporting() {
+        reportScheduledTask = scheduled.scheduleAtFixedRate(this::report, calculateInitialDelay(), REPORT_INTERVAL_MINUTES, MINUTES);
+    }
+
+    private void reportOnceIfNeeded() {
+        Path disabledReportingFile = dataDirectory.resolve(DISABLED_REPORTING_FILE_NAME);
+        if (!disabledReportingFile.toFile().exists()) {
+            scheduled.schedule(() -> {
+                if (report()) {
+                    saveDisabledReportingFile();
+                }
+            }, 1, TimeUnit.HOURS);
+        }
+    }
+
+    private void saveDisabledReportingFile() {
+        try {
+            Path disabledReportingFile = dataDirectory.resolve(DISABLED_REPORTING_FILE_NAME);
+            Files.writeString(disabledReportingFile, LocalDateTime.now().toString());
+        } catch (IOException e) {
+            LOG.debug("Failed to save disabled reporting file: ", e);
+        }
+    }
+
+    private void deleteDisabledReportingFileIfExists() {
+        try {
+            Files.deleteIfExists(dataDirectory.resolve(DISABLED_REPORTING_FILE_NAME));
+        } catch (IOException e) {
+            LOG.debug("Failed to delete disabled reporting file: ", e);
+        }
+    }
+
+    private long calculateInitialDelay() {
+        int currentMinute = LocalDateTime.now().getMinute();
+        int scheduledMinute = abs(deploymentID.hashCode()) % REPORT_INTERVAL_MINUTES;
+
+        if (currentMinute > scheduledMinute) {
+            return 60 - currentMinute + scheduledMinute;
+        } else {
+            return scheduledMinute - currentMinute;
         }
     }
```

Comment:
This should assert that `REPORT_INTERVAL_MINUTES` is 60 so we have less of a chance to miss the change.